### ISSUE TITLE
pandas 2.1.0 breaks duckdb when using arrow and query contains bind parameters

### ISSUE THREAD

**Issue by cpcloud** [duckdb/duckdb#8735]
### What happens?

When executing a query using `duckdb` from Python that contains bind parameters, I am unable to convert the result to Arrow using pandas 2.1.0 (I realize that it came out today). Without bind parameters, the query works.

### To Reproduce

```
In [1]: import duckdb

In [2]: import pandas as pd

In [3]: pd.__version__
Out[3]: '2.1.0'

In [4]: duckdb.execute("select ? as a", (1,)).arrow()
---------------------------------------------------------------------------
InvalidInputException                     Traceback (most recent call last)
Cell In[4], line 1
----> 1 duckdb.execute("select ? as a", (1,)).arrow()

InvalidInputException: Invalid Input Error: Required module 'pandas.core.arrays.arrow.dtype' failed to import, due to the following Python exception:
ModuleNotFoundError: No module named 'pandas.core.arrays.arrow.dtype'
```

### OS:

Linux

### DuckDB Version:

0.8.1

### DuckDB Client:

Python

### Full Name:

Phillip Cloud

### Affiliation:

Voltron Data

### Have you tried this on the latest `main` branch?

I have tested with a release build (and could not test with a main build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Tishj** [comment]
But yea to provide some context
The parameters are the issue, because we do a type check, and for that import pandas
importing pandas also imports all submodules that we might need (which is maybe a little too eager)

So there are a couple things to do:
1. Don't import for type checking, if the module is not imported, then the object can't be of any of the types in the module.
2. Make the `dtypes` module not required, so failing to locate the module is not an error
3. (requires some more engineering effort) make module imports more granular, so we don't import all submodules upfront

**Comment by SzilvasiPeter** [comment]
I can also confirm this issue. The first time execution failed and the following executions are succeeding. The workaround for me was to go back to the `2.0.0` version of pandas.

**Comment by cpcloud** [comment]
Thank you! Really appreciate the fix :)

**Comment by SzilvasiPeter** [comment]
The problem is still present. It seems that the 2.1.0 pandas version is breaking something. Here are the actions to reproduce the problem:

```bash
pip install duckdb polars[pyarrow] pandas pytest
```

This is the output of the `pip list` command. Note the 2.1.0 pandas version.
```
Package         Version
--------------- ------------
colorama        0.4.6
duckdb          0.8.1
exceptiongroup  1.1.3
iniconfig       2.0.0
numpy           1.25.2
packaging       23.1
pandas          2.1.0
pip             23.2.1
pluggy          1.3.0
polars          0.19.1
pyarrow         13.0.0
pytest          7.4.1
python-dateutil 2.8.2
pytz            2023.3.post1
setuptools      65.5.0
six             1.16.0
tomli           2.0.1
tzdata          2023.3
```

Now, create the `queries.py` script which contains two functions. It injects a DuckDB connection for easier unit testing.
```
import duckdb
import polars as pl


def fetch_data_from(con: duckdb.DuckDBPyConnection, table: str) -> pl.DataFrame:
    query = str.format("SELECT * FROM {0}", table)
    return con.execute(query).pl()


def last_nrow_from(
    con: duckdb.DuckDBPyConnection, table: str, limit: int = 7
) -> pl.DataFrame:
    return fetch_data_from(con, table).tail(limit)
```

Then, add unit test cases in the `test_queries.py` script. It executes three testcases.
```
import duckdb
import polars as pl
from polars.testing import assert_frame_equal

from queries import fetch_data_from, last_nrow_from


def test_fetch_data_from() -> None:
    expected_df = pl.DataFrame({"col1": [1, 2, 3], "col2": ["one", "two", "three"]})
    table_name = "test_table"

    with duckdb.connect() as con:
        con.register(table_name, expected_df)

        result_df = fetch_data_from(con, table_name)

        assert_frame_equal(result_df, expected_df)

def test_top_nrow_from_default_limit() -> None:
    table_name = "test_table"
    default_limit = 7

    expected_df = pl.DataFrame({"column": [1, 2, 3, 4, 5, 6, 7, 8]})

    with duckdb.connect() as con:
        con.register(table_name, expected_df)

        result_df = last_nrow_from(con, table_name)

        assert_frame_equal(result_df, expected_df.tail(default_limit))


def test_top_nrow_from_custom_limit() -> None:
    table_name = "test_table"
    custom_limit = 3

    expected_df = pl.DataFrame({"column": [1, 2, 3, 4, 5, 6, 7, 8]})

    with duckdb.connect() as con:
        con.register(table_name, expected_df)

        result_df = last_nrow_from(con, table_name, custom_limit)

        assert_frame_equal(result_df, expected_df.tail(custom_limit))
```

If you run `pytest -v` then **the second test cases will always fail**. No matter the order of the test cases, the second test will fail.

The workaround is to downgrade the pandas package to 2.0.0 version.
```
pip install --force-reinstall pandas==2.0.0
pytest -v
```

All of the test cases are passed successfully after downgrading the pandas package.

**Comment by Mytherin** [comment]
Note that this fix is not in v0.8.1 but only in the latest pre-release. Try `pip install duckdb --pre --upgrade`.

**Comment by edgarrmondragon** [comment]
Can confirm that this is patched in `0.8.2.dev4376`
